pkg/secretlock/secretsplitlock/secret_split_lock.go

```diff
@@ -59,15 +68,37 @@ func New(secret []byte, params *HubAuthParams, options ...Option) (secretlock.Se
 		options[i](opts)
 	}
 
+	doOnce.Do(func() {
+		opts.Logger.Infof("Initializing cache...")
+		cache = gcache.New(0).
+			LoaderFunc(func(key interface{}) (interface{}, error) {
+				uriAndToken := strings.Split(fmt.Sprintf("%s", key), "&")
+
+				return fetchSecretShare(uriAndToken[0], uriAndToken[1], opts.HTTPClient, opts.Logger)
```

Comment:
`opts.HTTPClient` and `opts.Logger` are registered for `doOnce.Do` and therefore do not change for subsequent calls. This is why unit-tests with custom http client are commented out.